### PR TITLE
Add `allowClicksThruHole` prop & step prop

### LINKED ISSUE #189: Add option to control allowing clicking in overlay hole
Clicking in the hole does not seem to trigger the "hole:click" callback.

I suspect it is related to #157 

The "hole:click" callback does get triggered in this PR for IanVS's fork when the new prop `allowClicksInHole` is set to false.
 https://github.com/nutshellcrm/react-joyride/pull/17

### PR DESCRIPTION
**Pull request by IanVS**
Fixes #189 

## Description

This fixes the breaking change which was introduced by https://github.com/gilbarbara/react-joyride/pull/157.  

It restores the behavior of version 1.7.0 of simply firing a callback when the hole is clicked, unless the prop `allowClicksThruHole` is set to `true` on the Joyride or on the step.

## To Test

Remove `allowClicksThruHole` from the step in the demo, and set `disableOverlay` to `false` and `debug` to `true`.  Verify that it is not possible to click the "Advance" button, and instead the callback `hole:click` is fired.

### CODE REVIEW

**Comment by IanVS** [comment]
Actually, I notice now the callback is still not firing.  Need to investigate.

**Comment by IanVS** [comment]
Ah, need to set `disableOverlay` to `false`, then it works.  I think we're good here.

**Comment by IanVS** [comment]
I thought it makes sense to allow it globally with a prop, so if that's how you want to use it for your entire joyride, you don't need to set it individually on every step.

**Comment by gilbarbara** [comment]
I find it confusing. I'd rather have this option on the step only as we talked before.

File: README.md [hunk 1]
```diff
@@ -134,6 +134,8 @@ You can change the initial options passing props to the component.
 
 **showOverlay** {bool}: Display an overlay with holes above your steps (for tours only). Defaults to `true`
 
+**allowClicksThruHole** {bool}: Allow mouse clicks and touch events within overlay hole, and prevent `hole:click` callback from being sent.  Defaults to `false`
```
Comment (gilbarbara) [inline README.md]: I think is better to use `mouse and touch events` instead of just mention clicks.

Comment (IanVS) [inline README.md]: 😕 I did say "mouse clicks and touch events"

**Pull request merged**